rllib/benchmarks/torch_compile/README.md

```diff
@@ -0,0 +1,76 @@
+
+
+# Benchmarks
+
+We conducted a comperhensive benchmark with this feature. 
+
+## Inference
+For the benchmarking metric, we compute the inverse of the time it takes to run `forward_exploration()` of the RLModule. We have conducted this benchmark on the default implementation of PPO RLModule under different hardware settings, torch versions, dynamo backends and modes, as well as different batch sizes. Here is a high-level summary of our findings:
+
+| Hardware | PyTorch Version | Speedup (%) | Backend + Mode           |
+|----------|----------------|-------------|--------------------------|
+| CPU      | 2.0.1          | 33.92       | ipex + default           |
+| CPU      | 2.1.0 nightly  | x           | ipex + default           |
+| T4       | 2.0.1          | 14.05       | inductor + reduce-overhead|
+| T4       | 2.1.0 nightly  | 15.01       | inductor + reduce-overhead|
+| V100     | 2.0.1          | 92.43       | inductor + reduce-overhead|
+| V100     | 2.1.0 nightly  | 85.71       | inductor + reduce-overhead|
+| A100     | 2.0.1          | x           | inductor + reduce-overhead|
+| A100     | 2.1.0 nightly  | 156.66      | inductor + reduce-overhead|
+
+
+For detailed benchmarks checkout [this google doc](https://docs.google.com/spreadsheets/d/1O7_vfGRLV7JfsClXO6stTg8snxghDRRHYBrR3f47T94/edit#gid=0). Here is the benchmarking code: [./run_inference_bm.py](./run_inference_bm.py). You can run the benchmark yourself as well:
+
+```bash
+./run_all_inference_bms.sh -bs <batch_size> --backend <dynamo_backend> --mode <dynamo_mode>
+```
+
+### Some meta level comments
+1. The performance improvement depends on many factors including the neural network architecture used, the batch size during sampling, the backend, the mode, the torch version, and many other things. The best way to optimize this is to first get the non-compiled workload learning and then do a hyper-parameter tuning on torch compile parameters on different hardwares.
+
+2. For CPU inference use the recommended inference only backends: `ipex` and `onnxrt`.
+
+3. The speedups are more significant on more modern architectures such as A100s compared to older ones like T4.
+
+4. Torch compile is still evolving. We noticed significant differences between the 2.0.1 release and the 2.1 nightly release. Therefore, it is important to take this into account during benchmarking your own workloads.
+
+
+## Exploration
+
+In RLlib, you can now set the configuration so that the compiled module is used during sampling of an RL agent training process. By default the rollout workers run on CPU, therefore it is recommended to use the `ipex` or `onnxrt` backend. Having said that, you can still choose run the sampling part on GPUs as well by setting `num_gpus_per_worker` in which case other backends can be used as well.
```

Comment:
```suggestion
In RLlib, you can now set the configuration so that the compiled module is used during sampling of an RL agent training process. By default, the rollout workers run on CPU, therefore it is recommended to use the `ipex` or `onnxrt` backend. Having said that, you can still choose to run the sampling part on GPUs as well by setting `num_gpus_per_worker` in which case other backends can be used as well.
```